pkg/redis/redis.go

```diff
@@ -110,15 +116,34 @@ type FailoverConfig struct {
 	MasterName string   `name:"master-name" description:"Redis Sentinel master name"`
 }
 
-func dial(ctx context.Context, network, addr string) (net.Conn, error) {
+func (c Config) dial(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
How about
```suggestion
func (c Config) makeDial(ctx context.Context, network, addr string) (func(net.Conn, error), error) {
```

Instead, which `newRedisClient` would call?
it's a very straight-forward change, which fits directly into current implementation and would allow us to reuse TLS config and initialization as well as fix the issue below